build_and_run_tests.sh

```diff
@@ -1,11 +1,16 @@
 set -x 
 
+#clean
+bazel clean
+
 #build
 bazel build --cxxopt='-std=c++14' //src/main:receive_main
 bazel build --cxxopt='-std=c++14' //src/main:send_main
 
 #bazel test --cxxopt='-std=c++14' --linkopt="-pthread" --test_output=all //tests:GoogleTest
-bazel run --cxxopt='-std=c++14' //tests:GoogleTest > test-log.txt
+bazel run --cxxopt='-std=c++14' --linkopt="-pthread" //tests:GoogleTest > test-log.txt
```

Comment:
I recommend you to end the name with .log instead of .txt It makes more sence and some editors put the right color on the test results, it gives a better readability.